WordPress/UITests/Utils/XCTest+Extensions.swift

```diff
@@ -95,6 +95,14 @@ extension XCTestCase {
         return paragraph
     }
 
+    /// Workaround for the black loading screen on iOS 18.
+    /// The screen shows up only on the first app launch, after 20-30 seconds.
+    public func allowForLoadingScreenInCI(_ app: XCUIApplication = XCUIApplication()) {
+        if ProcessInfo.processInfo.environment.keys.contains("BUILDKITE_BUILD_NUMBER") {
+            sleep(30)
```

Comment:
Is there a way we could instead add an XCUITest wait with a long timeout?

Reply:
That would be perfect. I actually tried to do that when handling this issue for Day One several weeks ago.
However, I couldn’t find a way to determine whether the loading screen was displayed. I couldn’t find a locator for it, nor any indication that something was being shown on top of the app. The app’s elements remain hittable all the time, and the framework even attempts to tap them.

Another challenge is that this loading screen can appear at any moment during the test, so we’d need to set a long timeout for essentially every element across multiple screens, which is also not ideal. =/

Reply:
What is a loading screen?

Reply:
Sorry for not providing all the context upfront. This is the loading screen I was referring to:

|  |  |
| - | - |
| <video width="300" src="https://github.com/user-attachments/assets/10242a2d-010c-43bf-81cb-15f8dc9877be"> | <video width="300" src="https://github.com/user-attachments/assets/a1f92475-fe51-40bd-82b2-de8c25167a6b"> |

The good news is that we might not need this piece of code here. I went through several recent runs, and, unlike the other apps above, it looks like the Jetpack app is not being impacted by it. They're on the same Xcode and iOS versions, the only difference I noticed was the VM images. Jetpack CI uses  `xcode-16.1` while the others use `xcode-16.1-v4`. I'll remove this workaround and observe how tests behave in the wild.

Reply:
Removed in [`5f5de6a`](https://github.com/wordpress-mobile/WordPress-iOS/pull/23748/commits/5f5de6a630dfa7c87cee08f0af9d07dbdaf1c226).

Reply:
When I've been working on these tests, I noticed that Xcode takes a massive performance hit when spinning simulators up. It's usually significantly faster to run tests if the required simulators are already available. I wish there was a way to spin all the required simulators, wait until they fully load, and only then run the tests. It seems that if you run tests in parallel with simulators being launched, it inevitably leads to issues.